File: Game Theory/Q2. The Ancient Scroll Hunt/jayantigoyal1--Q2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin>>n;
    vector<int> a(n);
    for(int i=0; i<n; i++){
        cin>>a[i];
    }
    int left=0, right=n-1;
    int indy=0, lara=0;
    bool flag=true;
    while(left<=right){
        int chosen;
        if(a[left]>a[right]){
            chosen=a[left];
            left++;
        }
        else{
            chosen=a[right];
            right--;
        }
        if(flag) indy+=chosen;
        else lara+=chosen;
        flag = !flag;
    }
    cout<<indy<<" "<<lara<<endl;
    return 0;
}

File: Game Theory/Q2. The Ancient Scroll Hunt/jayantigoyal1--Q2.py
n = int(input())
a = list(map(int, input().split()))

left = 0
right = n - 1
indy = 0
lara = 0
flag = True

while left <= right:
    if a[left] > a[right]:
        chosen = a[left]
        left += 1
    else:
        chosen = a[right]
        right -= 1

    if flag:
        indy += chosen
    else:
        lara += chosen

    flag = not flag

print(indy, lara)